### ISSUE TITLE
[require-atomic-updates] has too many false positives

### ISSUE THREAD

**Issue by fabiospampinato** [eslint/eslint#12423]
**Tell us about your environment**

* **ESLint Version:** 6.5.1
* **Node Version:** 10.15.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**

`@typescript-eslint/parser`

**Please show your full configuration:**

[Here](https://pastebin.com/Ni7HyKHL) it is, just uncomment the "require-atomic-updates" rule.

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
async fn ( note ) {
	let stat;
	if ( !stat ) stat = await File.stat ( note.filePath );
	note.metadata.created = stat ? stat.birthtime : new Date ();
}
```

```bash
npx eslint src --ext '.ts, .tsx'
```

**What did you expect to happen?**

No error detected.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint doesn't like my assignment, according to the rules of this rule:

> A variable or property is reassigned to a new value which is based on its old value

I don't think this is the case, the new value doesn't depend on the previous value.

> A yield or await expression interrupts the assignment after the old value is read, and before the new value is set.

Kind of, but `stat` is a locally scoped variable and by using `await` I'm pausing the entire function so nothing else could mutate the value assigned to `stat` while I'm awaiting.

**Are you willing to submit a pull request to fix this bug?**

I'm not too familiar with the AST, but with some pointers perhaps I can help.

**Comment by Mossop** [comment]
I'm seeing similar issues, maybe this is an issue with the TypeScript parser?

I get the error flagged just assigning an enum: https://github.com/FractalBrew/media-metadata/blob/master/src/ifd.ts#L244

**Comment by JakeRowsell89** [comment]
It may be related to `await` as opposed to TypeScript, could it be a duplicate of this: https://github.com/eslint/eslint/issues/11954

**Comment by platinumazure** [comment]
I'm going to go ahead and close this as a duplicate of #11899. Please follow that issue for updates. Thanks!
